include/z64cutscene.h

```diff
@@ -4,188 +4,867 @@
+} CsChooseCreditsScene;
+
+typedef union {
+    struct {
+        /* 0x0 */ u16 type;
+        /* 0x2 */ u16 startFrame;
+        /* 0x4 */ u16 endFrame;
+    };
+    s32 _words[2];
+} CsCmdMotionBlur; // size = 0x8
+
+typedef enum {
+    /* 1 */ CS_MOTION_BLUR_ENABLE = 1, // enable instantly
+    /* 2 */ CS_MOTION_BLUR_DISABLE // disable gradually
+} CsMotionBlurType;
+
+typedef union {
+    struct {
+        /* 0x0 */ u16 unused0;
+        /* 0x2 */ u16 startFrame;
+        /* 0x4 */ u16 endFrame; // unused
+    };
+    s32 _words[2];
+} CsCmdSfxReverbIndexTo2; // size = 0x8
+
+typedef union {
+    struct {
+        /* 0x0 */ u16 unused0;
+        /* 0x2 */ u16 startFrame;
+        /* 0x4 */ u16 endFrame; // unused
+    };
+    s32 _words[2];
+} CsCmdSfxReverbIndexTo1; // size = 0x8
+
+typedef union {
+    struct {
+        /* 0x0 */ u16 type;
+        /* 0x2 */ u16 startFrame;
+        /* 0x4 */ u16 endFrame;
+    };
+    s32 _words[2];
+} CsCmdTransition; // size = 0x8
+
+typedef enum {
+    /*  1 */ CS_TRANS_GRAY_FILL_IN = 1, // has hardcoded sounds for some scenes
+    /*  2 */ CS_TRANS_BLUE_FILL_IN,
+    /*  3 */ CS_TRANS_RED_FILL_OUT,
+    /*  4 */ CS_TRANS_GREEN_FILL_OUT,
+    /*  5 */ CS_TRANS_GRAY_FILL_OUT,
+    /*  6 */ CS_TRANS_BLUE_FILL_OUT,
+    /*  7 */ CS_TRANS_RED_FILL_IN,
+    /*  8 */ CS_TRANS_GREEN_FILL_IN,
+    /*  9 */ CS_TRANS_TRIGGER_INSTANCE, // used with `TRANS_MODE_INSTANCE_WAIT`
+    /* 10 */ CS_TRANS_BLACK_FILL_OUT,
+    /* 11 */ CS_TRANS_BLACK_FILL_IN,
+    /* 12 */ CS_TRANS_GRAY_TO_BLACK,
+    /* 13 */ CS_TRANS_BLACK_TO_GRAY
+} CutsceneTransitionType;
+
+typedef union {
+    struct {
+        /* 0x0 */ u16 textId; // can also be an ocarina action for `CS_TEXT_OCARINA_ACTION`
+        /* 0x2 */ u16 startFrame;
+        /* 0x4 */ u16 endFrame;
+        /* 0x6 */ u16 type;
+        /* 0x8 */ u16 altTextId1;
+        /* 0xA */ u16 altTextId2;
+    };
+    s32 _words[3];
+} CsCmdText; // size = 0xC
+
+typedef enum {
+    /* -1 */ CS_TEXT_TYPE_NONE = -1,
+    /*  0 */ CS_TEXT_TYPE_DEFAULT,
+    /*  1 */ CS_TEXT_TYPE_1,
+    /*  2 */ CS_TEXT_OCARINA_ACTION,
+    /*  3 */ CS_TEXT_TYPE_3,
+    /*  4 */ CS_TEXT_TYPE_BOSSES_REMAINS, // use `altText1` in the Giant Cutscene if all remains are already obtained
+    /*  5 */ CS_TEXT_TYPE_ALL_NORMAL_MASKS
+} CutsceneTextType;
+
+typedef union {
+    struct {
+        /* 0x0 */ u16 type;
+        /* 0x2 */ u16 startFrame;
+        /* 0x4 */ u16 endFrame;
+        /* 0x6 */ u8 intensity;
+        /* 0x7 */ u8 decayTimer;
+        /* 0x8 */ u8 decayStep;
+    };
+    s32 _words[3];
 } CsCmdRumble; // size = 0xC
 
-typedef struct {
-    /* 0x0 */ u16 unk0;
-    /* 0x2 */ u16 startFrame;
-    /* 0x4 */ u16 endFrame;
-    /* 0x6 */ Color_RGB8 color;
-    /* 0x9 */ UNK_TYPE1 pad9[0x3];
+typedef enum {
+    /* 1 */ CS_RUMBLE_ONCE = 1, // rumble once when startFrame is reached
+    /* 2 */ CS_RUMBLE_REPEATED // rumble every 64 frames between startFrame and endFrame
+} CutsceneRumbleType;
```

Comment:
havent looked that deep into this, but maybe pulse instead of repeat? every 64 frames is pretty decently spaced out